kong/plugins/ldap-acl/handler.lua

```diff
@@ -0,0 +1,41 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local BindConsumer = require "kong.plugins.ldap-acl.bind_consumer"
+local cache = require "kong.tools.database_cache"
+local responses = require "kong.tools.responses"
+local singletons = require "kong.singletons"
+
+local LdapAclHandler = BasePlugin:extend()
+
+function LdapAclHandler:new()
+  LdapAclHandler.super.new(self, "ldap-acl")
+end
+
+function LdapAclHandler:access(conf)
+  LdapAclHandler.super.access(self)
+
+  local ldap_authorization = ngx.ctx.authenticated_credential
+
+  if not ldap_authorization then
+    return responses.send_HTTP_UNAUTHORIZED()
+  end
+
+  if not ldap_authorization.username then
+    return responses.send_HTTP_UNAUTHORIZED("Unkown username")
```

Comment:
you're providing info to a potential attacker. Just respond, don't provide the reason why. Check the code, there are more of those.